android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.fxsuggest
+
+import android.content.res.Resources
+import mozilla.appservices.suggest.Suggestion
+import mozilla.appservices.suggest.SuggestionQuery
+import mozilla.components.concept.awesomebar.AwesomeBar
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.ktx.kotlin.toBitmap
+import java.util.UUID
+
+/**
+ * An [AwesomeBar.SuggestionProvider] that returns Firefox Suggest search suggestions.
+ */
+class FxSuggestSuggestionProvider(
+    private val resources: Resources,
+    private val loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
+    private val includeSponsoredSuggestions: Boolean,
+    private val includeNonSponsoredSuggestions: Boolean,
+    private val suggestionsHeader: String? = null,
+) : AwesomeBar.SuggestionProvider {
```

Comment:
I think it would be helpful to add kdoc `@param` entries for all of these as well (or at least the non-boolean types)

Reply:
Oops—done! 📝